examples/async2-passthru.rs

```diff
@@ -0,0 +1,447 @@
+
+    /// Poll for the packet event.
+    pub fn poll_packet_event(&mut self, cx: &mut Context) -> Poll<Result<()>> {
+        if self.ready.swap(false, Ordering::Relaxed) {
+            Poll::Ready(Ok(()))
+        } else {
+            self.waker.register(cx.waker());
+            Poll::Pending
+        }
+    }
+}
+
+impl Future for Win32EventFuture {
+    type Output = Result<()>;
+
+    /// Poll the future to check if the event is ready.
+    fn poll(self: std::pin::Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+        Pin::into_inner(self).poll_packet_event(cx)
+    }
+}
+/// Win32 event notifications
+struct Win32EventNotification {
+    win32_event: HANDLE,               // The Win32 event handle.
+    wait_object: HANDLE,               // The wait object handle.
+    callback: *mut Win32EventCallback, // A pointer to the Win32 event callback function.
+}
+
+impl std::fmt::Debug for Win32EventNotification {
+    /// Implementing the Debug trait for the Win32EventNotification struct.
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "Win32EventNotification: {:?}", self.wait_object)
+    }
+}
+
+type Win32EventCallback = Box<dyn Fn(BOOLEAN) + Send>; // A type alias for the Win32 event callback function.
+
+impl Win32EventNotification {
+    /// Register for Win32 event notifications.
+    fn new(win32_event: HANDLE, cb: Win32EventCallback) -> Result<Self> {
+        // Defining the global callback function for the Win32 event.
+        unsafe extern "system" fn global_callback(caller_context: *mut c_void, time_out: BOOLEAN) {
+            (**(caller_context as *mut Win32EventCallback))(time_out)
+        }
+
+        let callback = Box::into_raw(Box::new(cb)); // Creating a raw pointer to the callback function.
+        let mut wait_object: HANDLE = HANDLE(0);
+
+        // Registering for Win32 event notifications.
+        let rc = unsafe {
+            RegisterWaitForSingleObject(
+                &mut wait_object,
+                win32_event,
+                Some(global_callback),
+                Some(callback as *const c_void),
+                INFINITE,
+                WT_EXECUTEINWAITTHREAD,
+            )
+        };
+
+        // Check if the registration was successful.
+        if rc.as_bool() {
+            Ok(Self {
+                callback,
+                win32_event,
+                wait_object,
+            })
+        } else {
+            Err(unsafe { GetLastError() }.into())
+        }
+    }
+}
+
+impl Drop for Win32EventNotification {
+    /// Implementing the Drop trait for the Win32EventNotification struct.
+    fn drop(&mut self) {
+        unsafe {
+            // Deregistering the wait object.
+            if !UnregisterWaitEx(self.wait_object, self.win32_event).as_bool() {
+                //log::error!("error deregistering notification: {}", GetLastError);
+            }
+            drop(Box::from_raw(self.callback)); // Dropping the callback function.
```

Comment:
No need to explicitly call `drop`
```suggestion
            Box::from_raw(self.callback); // Dropping the callback function.
```

Reply:
Yes, but without drop, compiler complains `warning: unused return value of `Box::<T>::from_raw` that must be used`. Instead, I can use `let _ = Box::from_raw(self.callback);`. Which way is preferred?